docs/docs/clients.md

```diff
@@ -0,0 +1,246 @@
+
+Drand randomness is available over HTTP, [gRPC](https://grpc.io/) and [libp2p PubSub](https://docs.libp2p.io/concepts/publish-subscribe/), but availability of transports other than HTTP in client libraries is implementation dependent.
+
+Choose a drand client library that matches the language in which your application is written:
+
+- [Go](#go)
+- [JS](#js)
+
+Language not listed? You can make raw requests to the [HTTP API](/docs/http-api/). If you have created a client library or know of a client library not listed here, [please let us know by opening an issue](https://github.com/drand/website/issues/new).
+
+## Go
+
+### Install
+
+```sh
+go get github.com/drand/drand
+```
+
+### Usage
+
+[API reference on pkg.go.dev](https://pkg.go.dev/github.com/drand/drand/client)
+
+First import the base client, which wraps clients operating over different transports (HTTP, gRPC, libp2p PubSub etc.) with configured aggregation, caching, and retry logic:
+
+```go
+package main
+
+import (
+    "context"
+    "encoding/hex"
+    "fmt"
+    "github.com/drand/drand/client"
+)
+
+var chainHash, _ = hex.DecodeString("TODO-LOE-CHAIN-HASH")
+
+func main() {
+    c, err := client.New(
+        client.From(/* add concrete client implementations here (see below) */),
+        client.WithChainHash(chainHash),
+    )
+
+    // e.g. use the client to get the latest randomness round:
+    r := c.Get(context.Background(), 0)
+
+    fmt.Println(r.Round(), r.Randomness())
+}
+```
+
+The `From` option allows you to specify clients that work over particular transports. HTTP, gRPC and libp2p PubSub clients are provided. Note that you are not restricted to just one client. You can use multiple clients of the same type or of different types. The base client will periodically "speed test" it's clients, failover, cache results and aggregate calls to `Watch` to reduce requests.
+
+::: warning
+When using the client you _should_ use the `WithChainHash` or `WithChainInfo` option in order for your client to validate the randomness it receives is from the correct chain. You _may_ use the `Insecurely` option to bypass this validation but it is not recommended.
+:::
+
+#### HTTP
+
+The HTTP client uses the [JSON HTTP API](/docs/http-api/) to fetch randomness. Watching is implemented by polling the endpoint at the expected round time. The [League of Entropy](https://blog.cloudflare.com/league-of-entropy/) provides a set of public HTTP API endpoints, which are listed in the [HTTP API reference](/docs/http-api/).
+
+```go
+package main
+
+import (
+    "context"
+    "encoding/hex"
+    "fmt"
+    "github.com/drand/drand/client"
+    "github.com/drand/drand/client/http"
+)
+
+const url = "https://drand.cloudflare.com"
+
+var chainHash, _ = hex.DecodeString("TODO-LOE-CHAIN-HASH")
```

Comment:
Note this needs to be filled in - the currently running version of drand does not expose this hash on the `/info` endpoint so it needs to be generated from the group info.

Reply:
the current version semantics of e.g. latest are also different, so until there's an upgrade of the node software i don't think this is relevant

Reply:
👍 I'll leave as a TODO for now until the upgrade and open an issue to track.

Reply:
https://github.com/drand/website/issues/52